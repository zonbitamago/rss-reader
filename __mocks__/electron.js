const remote ={
  app : {
    getPath :function(str){return '.'}
  }
}

module.exports = {
  require: jest.genMockFunction(),
  match: jest.genMockFunction(),
  app: jest.genMockFunction(),
  remote: remote,
  dialog: jest.genMockFunction()
};
