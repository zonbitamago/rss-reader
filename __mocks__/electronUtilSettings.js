export const electronUtil = {
  closeApp: () => {
    console.log("closeApp");
  },
  minimizeApp: () => {
    console.log("minimizeApp");
  },
  openBrowser: url => {
    console.log("openBrowser");
    console.log(url);
  }
};
