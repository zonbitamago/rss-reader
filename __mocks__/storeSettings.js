import { resolve } from "bluebird-lst";
import moment from "moment";

export let store = {
  ItemStore: {
    items: [],
    loading: false,
    updateDate: moment("2012-06-20 09:12:34").format("HH:mm:ss"),
    updateDuration: 5,
    timerId: "dummy",
    add: () => console.log("add"),
    getSettings: () => console.log("getSettings"),
    setSetting: () => console.log("setSetting"),
    setTimer: () => console.log("setTimer")
  },
  FeedListStore: {
    feedList: [
      { name: "google", url: "https://www.google.co.jp/" },
      { name: "yahoo", url: "https://www.yahoo.co.jp/" }
    ],
    getFeedList: () => console.log("getFeedList"),
    setFeedList: () => {
      console.log("setFeedList");
      return new Promise(resolve).then(() => {
        return true;
      });
    },
    deleteFeedList: () => console.log("deleteFeedList")
  }
};
