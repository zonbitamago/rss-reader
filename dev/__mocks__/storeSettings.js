export let store = {
  ItemStore: { items: [], add: () => console.log("add") },
  SettingStore: {
    updateDuration: 5,
    getSettings: () => console.log("getSettings"),
    setSetting: () => console.log("setSetting")
  },
  RssListStore: {
    rssList: [
      { name: "google", url: "https://www.google.co.jp/" },
      { name: "yahoo", url: "https://www.yahoo.co.jp/" }
    ],
    getRssList: () => console.log("getRssList"),
    setRssList: () => console.log("setRssList"),
    deleteRssList: () => console.log("deleteRssList")
  }
};
