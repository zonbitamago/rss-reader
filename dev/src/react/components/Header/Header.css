.Header {
  background-color: #039be5;
  width: 100%;
  height: 3em;
  color: white;
  /* display: grid;
  grid-template-columns: 3em 1fr 6em; */
  display: flex;
  justify-content: space-between;
}
.Header > div {
  height: inherit;
}
.Header-Left {
}
.Header-Center {
  display: flex;
  justify-content: center;
  align-items: center;
}
.Header-Center > div {
  margin: 0px 5px 0px 5px;
}

.Header-Left > .Icon {
  cursor: pointer;
}

.Header-Right {
  display: flex;
  align-items: center;
}

.Header-Right > div {
  margin-right: 0.5em;
}
