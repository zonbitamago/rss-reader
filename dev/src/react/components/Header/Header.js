import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Header.css";
import Icon from "../Icon/Icon";
import Home from "@material-ui/icons/Home";
import RssFeed from "@material-ui/icons/RssFeed";
import Twitter from "react-icons/lib/ti/social-twitter";
import Typography from "material-ui/Typography";

class Header extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="Header">
        <div className="Header-Left">
          <Icon>
            <Home />
          </Icon>
        </div>
        <div className="Header-Center">
          <Icon>
            <RssFeed />
          </Icon>
          <Typography color="inherit" variant="subheading">
            +
          </Typography>
          <Icon>
            <Twitter />
          </Icon>
          <Typography color="inherit" variant="subheading">
            Feed
          </Typography>
        </div>
        <div className="Header-Right" />
      </div>
    );
  }
}

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
