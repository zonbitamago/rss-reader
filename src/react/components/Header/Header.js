import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Header.css";
import Icon from "../Icon/Icon";
import Home from "@material-ui/icons/Home";
import RssFeed from "@material-ui/icons/RssFeed";
import Twitter from "react-icons/lib/ti/social-twitter";
import Typography from "@material-ui/core/Typography";
import $ from "jquery";

class Header extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="Header">
        <div className="Header-Left">
          <Icon>
            <Home
              onClick={() => {
                $(".main").animate({ scrollTop: 0 });
              }}
            />
          </Icon>
        </div>
        <div className="Header-Center">
          <Icon>
            <RssFeed />
          </Icon>
          <Typography color="inherit" variant="subheading">
            +
          </Typography>
          <Icon>
            <Twitter />
          </Icon>
          <Typography color="inherit" variant="subheading">
            Feed
          </Typography>
        </div>
        <div className="Header-Right">
          <div>update : {this.props.store.ItemStore.updateDate}</div>
        </div>
      </div>
    );
  }
}

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
