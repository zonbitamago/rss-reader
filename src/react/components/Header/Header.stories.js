import React from "react";
import { storiesOf } from "@storybook/react";
import { store } from "../../../../__mocks__/storeSettings";

import Header from "./Header";

storiesOf("Organisms/Header", module).add("Header", () => (
  <Header store={store} />
));
