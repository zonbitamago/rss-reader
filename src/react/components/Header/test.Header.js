import React from "react";
import { shallow, mount } from "enzyme";
import $ from "jquery";
import { store } from "../../../../__mocks__/storeSettings";

import Header from "./Header";

jest.mock("jquery");

describe("Header", () => {
  let component, props;

  beforeEach(() => {
    props = { store: store };
    component = shallow(<Header {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });

  it("Home Icon Click", () => {
    $.mockImplementation(() => {
      return {
        animate: selector => {
          return;
        }
      };
    });
    var wrapper = mount(<Header {...props} />);

    wrapper.find(".Header-Left .Icon svg").simulate("click");
    expect($.mock.calls.length).toBe(1);
  });
});
