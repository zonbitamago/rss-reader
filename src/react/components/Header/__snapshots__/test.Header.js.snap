// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should 1`] = `
<div
  className="Header"
>
  <div
    className="Header-Left"
  >
    <Icon>
      <pure(Home)
        onClick={[Function]}
      />
    </Icon>
  </div>
  <div
    className="Header-Center"
  >
    <Icon>
      <pure(RssFeed) />
    </Icon>
    <WithStyles(Typography)
      color="inherit"
      variant="subheading"
    >
      +
    </WithStyles(Typography)>
    <Icon>
      <TiSocialTwitter />
    </Icon>
    <WithStyles(Typography)
      color="inherit"
      variant="subheading"
    >
      Feed
    </WithStyles(Typography)>
  </div>
  <div
    className="Header-Right"
  >
    <div>
      update : 
      09:12:34
    </div>
  </div>
</div>
`;
